### ISSUE TITLE
They're really cracking down on the IP bans

### ISSUE THREAD

**Issue by heffman92** [finndev/PokeBuddy#793]
I've tried multiple IPs and within 15-20 minutes it stops working. And half of the IPs don't even work from the bat. They're really cracking down lol.

**Comment by HenghaoZhou** [comment]
I've been running well for almost 2hrs with the latest version.

**Comment by heffman92** [comment]
I keep having to change my IP. Ugh

**Comment by curtiola** [comment]
Do you just release and renew or do you unplug the router?

**Comment by curtiola** [comment]
Also some companies like Comcast don't support dynamic IP and it won't change anyway

**Comment by heffman92** [comment]
VPN. And I tried resetting my router but the only thing that worked was VPN.

**Comment by heffman92** [comment]
Yeah I have Xfinity lol

**Comment by Rawr2134** [comment]
yea =\ im comcast here and it wont change, im looking around in my router config to see if there is a setting or if they have to set it on their end

**Comment by curtiola** [comment]
Ah, strange, the VPN you use in the same location you set pokebuddy to? They might have "region locked" it so to speak

**Comment by HenghaoZhou** [comment]
try #779

**Comment by heffman92** [comment]
Well I've been using Japan cause that's the only one that seems to work. I've tried other countries as well

**Comment by curtiola** [comment]
Is that where you set pokebuddy to? I would suspect that if the IP doesn't agree with the GPS it would be a red flag

**Comment by heffman92** [comment]
No. I'm in Minnesota so my pokebuddy is running in my state not far from me. But my VPN has all different countries locations. Japan seems to be the only one that works for me

**Comment by curtiola** [comment]
That could be the problem, try Japan (after allowing ample time to travel) or maybe try using proxifier or something similar if you don't want to wait a long time, I really don't know if the IP and gps locations are compared but I think it is

**Comment by marksalotlies** [comment]
I get one every 20 mins. I close down PokeBuddy and unplug my router. When I plug it back in and open up Pokebuddy it works again but it's annoying. (The incubator thing doesn't help at all)

**Comment by curtiola** [comment]
Have you tried running pokebuddy in an area near you IP location? Could just be that your IP and gps locations differ a lot, which would be obvious to Niantic

**Comment by marksalotlies** [comment]
I'm using a location about an hour away from me. This seems to have just started a few hours ago. There really aren't many stops anywhere closer... but I'll try for the sake of testing if that helps.

**Comment by curtiola** [comment]
An hour away really shouldn't be an issue with the gps/IP comparison

**Comment by curtiola** [comment]
I'm only speculating either way though lol

**Comment by catch3mAll** [comment]
I am also constantly getting IP banned. I put my pokebuddy exactly where the IP is and it works for a few mins and then I get IP ban.

**Comment by curtiola** [comment]
Have you renew the IP? If you just close it and reopen it you would still be banned, might not be instant, few minutes sounds right, some companies like Comcast don't support dynamic IP and it won't change, in that case you may need a VPN or proxifier

**Comment by catch3mAll** [comment]
I used VPN, changed it a few times with matching GPS locations, still IP ban after a few mins

**Comment by curtiola** [comment]
You sure it's an IP ban? Can't see pokestops on live location is a clear indicator (though sometimes that's server related)
If so then I'm not sure my theory holds up lol

**Comment by catch3mAll** [comment]
Pretty sure its IP ban, cause I cant see any pokestops. Also, changing IP works for a bit then it doesn't work, so this also seems like its an IP Ban.

**Comment by curtiola** [comment]
Yeah it sure does, curious as to how they detect it, I assumed it was multiple accounts sending the same sensor data from the same IP or gps/IP location differences that cause it, what else are they checking that might do this?

**Comment by catch3mAll** [comment]
Maybe other people are using the same VPN?

**Comment by curtiola** [comment]
Good theory, very possible if they're also using pokebuddy lol have you tried proxifier? I don't know if you can choose locations or anything with it, never used it, but others have recommended it

**Comment by marksalotlies** [comment]
I'm only using 1 account, 1 bot. I get a new IP everytime I unplug my router. They must have found a way to detect the bots that takes a few mins. So far, for me it's between 20-22 mins.

Running my bot closer to home isn't panning out because it runs out of stops so it just stands there saying no pokemon or pokestops can be found. lol

**Comment by curtiola** [comment]
Hm, I wonder what it could be, perhaps we need to have a +/- X km per hour variation, no person travels at a constant pace, or maybe this has something to do with some sort of device information, I really don't know lol

**Comment by 2stronkkk** [comment]
RIP BOTS 2K16- 2K16

**Comment by 2stronkkk** [comment]
dont worry boys. niantic programmer is here to save the world

**Comment by curtiola** [comment]
Don't start that again lol we're having an intelligent discussion xD

**Comment by catch3mAll** [comment]
Also, maybe they can detect VPNs. I just remote connected to my laptop at home and ran pokebuddy instead of doing this at work and it seems to be working.

**Comment by 2stronkkk** [comment]
i think the problem is ip location and gps location. need to be close or u will be banned. i am gonna check it when i got unbanned

**Comment by marksalotlies** [comment]
Potentially also an elevation variation. Or maybe a walking speed with decimals? There are so many things it could be. There's another bot app where this problem doesn't seem to be coming up in their issues. That app is waaayyy more complicated so maybe Pokebuddy can figure out what's missing here.

I'm not using a vpn at all.

**Comment by catch3mAll** [comment]
nvm, got IP Banned again, i hope someone figures out how they are detecting the bot soon

**Comment by 2stronkkk** [comment]
do u already test what i said catch? use ur ip and ur gps in the same location

**Comment by catch3mAll** [comment]
@2stronkkk my ip and gps location is pretty much the same location, within a few miles

**Comment by RoseCorpse** [comment]
I'm botting like a glove (insert Ace Ventura's meme here)

**Comment by RoseCorpse** [comment]
I'm from Venezuela, botting in Australia btw, and my bot is A-okay.

**Comment by RoseCorpse** [comment]
I LOVE YOU DEVS <3

**Comment by 2stronkkk** [comment]
i dont know then. so sad

**Comment by Sorata5000** [comment]
Are you using the most updated version of pokebuddy? I just got IP banned as well. Only one instance of pokebuddy open and then used nox. No pokestops showing up but when I take my phone off wifi and go into the app stops load up.

**Comment by marksalotlies** [comment]
Ok, i'm botting in my real location and I made it past 22 mins. BUT... i spend 80% of the time standing still waiting for the couple pokestops to activate again (there's essentially 0 pokemon). So I'm thinking it's one of 2 things... either the location matching with the IP is important or they are detecting us by using something about the interaction with pokemon/pokestops.

**Comment by RoseCorpse** [comment]
I LOVE YOU DEVS <3

**Comment by marksalotlies** [comment]
Then again I'm not moving so it could be speed/elevation. I wish the bot would wander back to the start when it ran out of pokestops instead of standing still.

**Comment by 2stronkkk** [comment]
rose we already know that.

**Comment by curtiola** [comment]
If you can see stops on your phone but not on pokebuddy then it's most likely not an IP ban (if you're using wifi)

**Comment by curtiola** [comment]
I now see that you said you took your phone off wifi, maybe altitude is a factor, central park would be 17, don't know other locations altitude and I'm not at home right now so I can't test this

**Comment by pokeymans1** [comment]
Oddly enough, I tried to bot today in my location. On my phone, I can go fight gyms and such, but my bot doesnt hit any pokestops, They are there, it just wont hit them. 

I tried going on a VPN, didnt work.
I tried resetting my laptop, didnt work.
I tried setting a static IP, didnt work.
I tried unchecked incubators, and one or the other checked, didn't work.
I tried redownloading and starting from scratch, and setting the bot close and exactly where I am, got banned in 5 minutes.
Phone works fine though (on 4G)

Anyone else experiencing the same issue? Getting banned no matter what. :(

**Comment by 2stronkkk** [comment]
like me

**Comment by curtiola** [comment]
Maybe it's another signature of some sort like marksalotlies suggested, could be something specific the bot interacts with, anyone notice a pattern in the logs? Maybe it's always after a pokemon is caught or something, or could it be walk speed? Above 25 and the app notifies you that you shouldn't play and drive and the bot might not be pressing the OK button or whatever lol try keeping it low? I usually run below 11

**Comment by Sorata5000** [comment]
Im still IP banned so I cant test it curiola. >.<

**Comment by RoseCorpse** [comment]
you guys are scaring me.

**Comment by pokeymans1** [comment]
Nothing really stood out when I got banned tbh. It was running around like normal. Picked up 5 pokemon. The only thing i noticed is that I get banned after 3-5 minutes or so. Nothing outstanding. It just happens.

I did go to Central Park in NY last night. Same thing happened. about 5 minutes in.. BOOM stops in place, does not walk, (if it does, very very slowly), won't do anything else.

**Comment by 2stronkkk** [comment]
i am testing right now. using ip and gps in the same location. pray for me

**Comment by pokeymans1** [comment]
May Helix Provide Guidance to @2stronkkk

**Comment by curtiola** [comment]
Huh, it's hard to say what causes it, could be something different in every situation for all I know lol I won't be home to test out these theories for at least a week sadly, it could be as simple as a pokemon transfer that does it, or evolving too quickly, since on the app it takes like 25 seconds, I have lots of things to test if it's not figured out in a week haha

**Comment by pokeymans1** [comment]
I'll keep trying, and just post everything I do here. I'll try and clean up my logs when I do.

**Comment by 2stronkkk** [comment]
well bot its working .farm pokestops but only get 5 pokemons in 8 minutes. its so low. before i got 20 in 2 min xd

**Comment by curtiola** [comment]
@2stronkkk also below 25km/hr walking speed? Also if you wouldn't mind posting the entries in the log where you become softbanned, I'd be ahppy

**Comment by pokeymans1** [comment]
@2stronkkk do you have any Pokemon in gyms?

**Comment by curtiola** [comment]
IP banned** not soft lol..

**Comment by 2stronkkk** [comment]
this is bullshit. 20k/h xD

**Comment by pokeymans1** [comment]
@curtiola I was using 5km/h walk speed on mine, if that helps. I also had pokemon in 2 Gyms.

**Comment by Skolzack** [comment]
I have 50km/hr walk speed, did 63k/hr yesterday and today dropped to 52k/hr. The map is much more slow, the actions too and idk why..

**Comment by 2stronkkk** [comment]
RESUME- my ip location is portuguese. when i use bot in US i got banned in 3 minutes. always. now i put bot in portugal i can play without problems. its important to know

**Comment by curtiola** [comment]
Yeah that helps, just wish there was an easy way to isolate the cause lol so many things to try, speed still might be one of the causes but it's definitely not the only one

**Comment by pokeymans1** [comment]
@2stronkkk might have a point there. I'll try a different country. I read somewhere that japan was working, but very slowly.

**Comment by 2stronkkk** [comment]
true. but its one of them for sure! gl boys my time is now. see u tomorrow

**Comment by curtiola** [comment]
Thanks @2stronkkk that is helpful, did you notice what the bot did just before being IP banned in the US?

**Comment by 2stronkkk** [comment]
before ip banned bot is farming 2/3 pokestops and then stop farming and stay in the same place

**Comment by curtiola** [comment]
OK ty, see ya tomorrow

**Comment by RoseCorpse** [comment]
I don't know if this will help, but I'll share with guys my configs (my bot is working fine, no ip ban at the moment) 15km/p farming in Sydney. Tasks:

- Catch Pokemon
- Farm Pokestops
- Recycling items
- Evolving pokemon (no lucky eggs)
- Use incubators
- Transfer bad pokemon.

I'm using Nox, not cellphone, same coordinates in bot and nox, I have 2 pokemons at gyms in my country, I teleport to Venezuela once a day just to check gyms/own gyms.

I guess that's all, I hope it helps.

**Comment by curtiola** [comment]
That helps, thanks, is your pokebuddy running the latest version or has it been running for a day or 2 without closing/updating? I'm curious if it was something to do with pokebuddy update last night or not

**Comment by RoseCorpse** [comment]
I forgot that point, I have not updated my pokebuddy since early morning, I just stop it, open nox, oh! and I see my pokebuddy last coordinate and I copy/paste it in nox for appearing at the same spot that was pokebuddy before I stop it.

**Comment by curtiola** [comment]
OK thank you, I don't think it was a pokebuddy update then, these problems began yesterday

**Comment by marksalotlies** [comment]
I moved my bot about half an hour from my house and it's been actively catching pokemon/spinning pokestops for about an hour. (the longest since this started happening) Someone else mentioned being near Central Park when they were banned... that's also where I was when I was banned. Could they be more strict about high bot traffic areas? I'm going to leave the bot offline overnight and try a totally different big city tomorrow to see if it helps.

**Comment by curtiola** [comment]
I don't think it has anything to do with specific areas, I think it would be too easy to get false positives and accidentally IP ban one of the people actually using their phone

**Comment by curtiola** [comment]
Unless they can somehow check if you're connected to a cell tower in the area for Internet or not, I don't know what info they have access to lol

**Comment by marksalotlies** [comment]
That's true... unless maybe they only monitor whatever it is they are catching us by in certain areas because of their server capacity issues. It could even be a pilot like how they are piloting the new "sightings" feature. 

But it's all just guessing! lol

**Comment by RoseCorpse** [comment]
@marksalotlies I don't think so, I'm training in Sydney and it's a very crowded area, really really crowded.

**Comment by curtiola** [comment]
Guessing is all I can do right now too lol I won't have a pc to check things out on for a while

**Comment by pokeymans1** [comment]
10 minutes into PokeBuddy in Japan, I got IP Banned. Lol. Perhaps its my account....

**Comment by curtiola** [comment]
Could somehow be account related, previously IP banned accounts might be under more strict guidelines or some shit, I have no idea haha

**Comment by curtiola** [comment]
I assume it's not account related though, it doesn't seem likely

**Comment by RoseCorpse** [comment]
Damn, I'll keep my ass only Sydney, won't teleport ATM.

**Comment by pokeymans1** [comment]
Yeah. I did use a different VPN. Did one out of Japan. Worked for a good 10 minutes. Then banned. Ill give it a rest for a bit, and try it again with the incubator trick in other threads.

**Comment by pokeymans1** [comment]
17 minutes and counting... unchecked recycle items... unchekced 1 incubator... seems to be working fine... new Japan IP

**Comment by curtiola** [comment]
Awesome, hope it lasts, let me know if it lasts over like an hour

**Comment by pokeymans1** [comment]
Will do, 30 minutes and counting. Still going strong. I'll post my full config if it reaches an hour.

**Comment by curtiola** [comment]
Perfect xD also mention what area your IP is in with the rest of the info please

**Comment by footekruk** [comment]
Logged in with my real account and getting a no gps error

**Comment by pokeymans1** [comment]
2 hours and 15 minutes in. Things are looking good.

Used Soft Ether VPN Client and picked a random Japanese VPN Tunnel.
Unchecked Use Incubators. Use main Incubator checked.
Walk Speed 4 km/h
Map Disneyland Tokyo
Unchecked Evolve
Unchecked Recycle Items (periodically check it to clear items)

37k exp / hr.

**Comment by pokeymans1** [comment]
LOL and im banned.. HAHAHAHA

**Comment by curtiola** [comment]
Still good info, thanks for posting results, any chance you remember exactly what the bot did when you got IP banned?

**Comment by pokeymans1** [comment]
@curtiola walking to another pokestop. It just happens.... -1:

**Comment by heffman92** [comment]
Yeah. The most I've gotten it to run is an hour. Gets IP blocked like every time. When u switch to my normal IP after not using it for a day it works for like 5 minutes then stops. So much work to keep connecting to other IPs
